drivers/hmis/lib/ce_workflows/ac/workflow_builder.rb

```diff
@@ -337,26 +388,36 @@ def build_provider_outcome_denial_review_loop(template:, ce_staff_swimlane:, pro
       # Denial Review 1 => Gateway => Decline OR send back to Provider Outcome
       denial_review_gateway_1 = CeWorkflows::Shared::CeBuilderUtils.create_gateway(template, 'denial_review_1')
       denial_review_task.connect_to!(denial_review_gateway_1)
-      denial_review_gateway_1.connect_to!(decline_event, condition: 'denial_review_decision = 1') # Accept Denial
+      denial_review_gateway_1.connect_to!(admin_decline_gateway, condition: 'denial_review_decision = 1') # Accept Denial
       denial_review_gateway_1.connect_to!(provider_outcome_task_2) # "Send back" to next attempt at Provider Outcome
 
       # Denial Review 2 => Gateway => Decline OR send back to Provider Outcome
       denial_review_gateway_2 = CeWorkflows::Shared::CeBuilderUtils.create_gateway(template, 'denial_review_2')
       denial_review_task_2.connect_to!(denial_review_gateway_2)
-      denial_review_gateway_2.connect_to!(decline_event, condition: 'denial_review_decision = 1') # Accept Denial
+      denial_review_gateway_2.connect_to!(admin_decline_gateway, condition: 'denial_review_decision = 1') # Accept Denial
       denial_review_gateway_2.connect_to!(provider_outcome_task_3) # "Send back" to next attempt at Provider Outcome
 
       # Denial Review 2 => Gateway => Decline. Cannot be "sent back" to Provider Outcome.
-      denial_review_task_3.connect_to!(decline_event)
+      denial_review_task_3.connect_to!(admin_decline_gateway)
 
       # Create Enrollment (Script) => Confirm Success Task
       create_enrollment_task.connect_to!(confirm_success_task)
 
-      # Confirm Success Task => Accept Event
-      confirm_success_task.connect_to!(decline_event, condition: 'move_forward = 0')
-      confirm_success_task.connect_to!(accept_event, condition: 'move_forward = 1')
+      # Create Enrollment (Script) => Change Provider Outcome *optional* task that re-triggers the Denial Review
+      create_enrollment_task.connect_to!(change_provider_outcome_task)
+      # Note that this re-introduces the truly circular workflow.
+      # It is theoretically possible for users to re-open a previously closed Denial Review step, like this:
+      # - Complete the Change Provider Outcome step
+      # - In the Denial Review step, select "Send Back"
+      # - In the Provider Outcome (Second Attempt) step, select "Accept/Enroll"
+      # - Complete the Change Provider Outcome step a second time
+      # - The Denial Review (1) step is re-opened, instead of Denial Review (Second Attempt).
+      change_provider_outcome_task.connect_to!(denial_review_task)
```

Comment:
Let's discuss, does this seem acceptable?

Reply:
If we wanted to avoid cycles, perhaps we could add an additional user task? Maybe the cycles are fine though